### PR TITLE
Fixes #160: Integrating topic controller into Concept Card

### LINKED ISSUE #160: UI Structure: ConceptCardFragment
This is tracking implementing the initial UI structure for the concept card fragment that's being fully implemented by #24.

### PR DESCRIPTION
**Pull request by jamesxu0**
## Explanation

Fixes #160: Integrates the topic controller into Concept Card. Concept Card uses data binding to bind the explanation and worked examples into text views (blocked on #26). Uses the binable recyclerview introduced in PR  #172, which has not been merged yet.

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is **assigned** to an appropriate reviewer.

### CODE REVIEW

File: app/src/main/java/org/oppia/app/recyclerview/BindableAdapter.kt
Comment (BenHenning) [inline app/src/main/java/org/oppia/app/recyclerview/BindableAdapter.kt]: Can you change this PR to diff against the branch in #172 so that these extra files aren't included?

File: app/src/main/java/org/oppia/app/topic/conceptcard/ConceptCardFragment.kt
Comment (BenHenning) [inline app/src/main/java/org/oppia/app/topic/conceptcard/ConceptCardFragment.kt]: Suggest defaulting to null and having another checkNotNull() to ensure the skill ID is passed to the concept card fragment. There are some cases when this could fail, but I think that's fine. If we lose the skill ID, we can't really recover.

Comment (BenHenning) [reply app/src/main/java/org/oppia/app/topic/conceptcard/ConceptCardFragment.kt]: Also with this change, we wouldn't need to import TEST_SKILL_ID_2 here (which we ought to not anyway--UI structure work should be close to "prod-ready" even if the domain controllers it depends on aren't yet ready).

File: app/src/main/java/org/oppia/app/topic/conceptcard/ConceptCardPresenter.kt
Comment (BenHenning) [inline app/src/main/java/org/oppia/app/topic/conceptcard/ConceptCardPresenter.kt]: I don't actually think we need to set this if it's also being set in the layout.

File: app/src/main/java/org/oppia/app/topic/conceptcard/testing/ConceptCardFragmentTestActivity.kt
Comment (BenHenning) [inline app/src/main/java/org/oppia/app/topic/conceptcard/testing/ConceptCardFragmentTestActivity.kt:8]: Please add documentation for this class, even if it's not detailed. Ditto below for its presenter.

File: app/src/main/java/org/oppia/app/topic/conceptcard/testing/ConceptCardFragmentTestActivity.kt
Comment (BenHenning) [inline app/src/main/java/org/oppia/app/topic/conceptcard/testing/ConceptCardFragmentTestActivity.kt]: Should be ``ConceptCardFragmentTestActivityPresenter``

File: app/src/main/res/layout/concept_card_fragment_test_activity.xml
Comment (BenHenning) [inline app/src/main/res/layout/concept_card_fragment_test_activity.xml]: Please add newline at end of file. Ditto elsewhere in this PR.

File: app/src/main/res/layout/concept_card_fragment_test_activity.xml
Comment (BenHenning) [inline app/src/main/res/layout/concept_card_fragment_test_activity.xml]: If this is the only child in the layout, maybe just use a FrameLayout instead. Constraints don't seem needed here.

File: app/src/main/res/layout/conceptcard_fragment.xml
Comment (BenHenning) [inline app/src/main/res/layout/conceptcard_fragment.xml:1]: Nit on file name: ``concept_card_fragment``

File: app/src/sharedTest/java/org/oppia/app/topic/conceptcard/ConceptCardFragmentTest.kt
Comment (BenHenning) [inline app/src/sharedTest/java/org/oppia/app/topic/conceptcard/ConceptCardFragmentTest.kt]: Nit: ``workedExamplesAreDisplayed``

File: app/src/sharedTest/java/org/oppia/app/topic/conceptcard/ConceptCardFragmentTest.kt
Comment (BenHenning) [inline app/src/sharedTest/java/org/oppia/app/topic/conceptcard/ConceptCardFragmentTest.kt:30]: Other tests that might be nice:
- Launching with a different skill ID correctly shows a different concept card
- Rotating/a configuration change continues to show the correct concept card

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks @jamesxu0! Generally this LGTM. I think you're blocked on #172 being submitted first, though, so I'll prioritize that work.

File: app/src/main/java/org/oppia/app/recyclerview/BindableAdapter.kt
Comment (jamesxu0) [reply app/src/main/java/org/oppia/app/recyclerview/BindableAdapter.kt]: I changed to diff against the branch, but the branch is pretty far behind develop, so there's a lot of other changes in here now too.

**Comment by jamesxu0** [comment]
@BenHenning PTAL. Add test cases for ConceptCard using ActivityTestRule (for screen rotation).

File: app/src/main/res/drawable/lesson_thumbnail_graphic_baker.xml
Comment (BenHenning) [inline app/src/main/res/drawable/lesson_thumbnail_graphic_baker.xml]: Aren't these already imported per https://github.com/oppia/oppia-android/blob/develop/app/src/main/res/drawable/lesson_thumbnail_graphic_baker.xml?

**Review by BenHenning — CHANGES_REQUESTED** [review]
Ah, I think that the base branch is a bit behind develop. I'll prioritize finishing that work tonight and then this PR can be updated to base off of develop to simplfiy the review.

Passing this back over to you @jamesxu0 to update the base branch once #172 is merged in.

**Comment by jamesxu0** [comment]
@BenHenning PTAL. I merged in the bindable recyclerview to this branch again, so only my changes show in the diff.

File: app/src/sharedTest/java/org/oppia/app/topic/conceptcard/ConceptCardFragmentTest.kt
Comment (BenHenning) [inline app/src/sharedTest/java/org/oppia/app/topic/conceptcard/ConceptCardFragmentTest.kt]: Does this work as-is or do you also need to call [recreate](https://developer.android.com/reference/androidx/test/core/app/ActivityScenario.html#recreate()) on the activity scenario? Did you verify via step debugging that the activity is recreated? Please update test accordingly to make sure that it does go through the rotation as expected.

**Review by BenHenning — APPROVED** [review]
Thanks @jamesxu0! I only had one concern around the configuration testing, but otherwise this LGTM!

**Pull request merged**
